openmmtools/alchemy.py

```diff
@@ -2502,6 +2630,283 @@ def check_energy_expression(custom_force, parameter):
+    if not bonded_force:
+        switch_string = (
+            "in_alchemical1 + in_alchemical2 - in_alchemical1*in_alchemical2" if do_annihilate else
+            # 1 if one or both particles are in the alchemical region, 0 otherwise
+            "in_alchemical1 + in_alchemical2 - 2*in_alchemical1*in_alchemical2"
+            # 1 if one and only one particle is in the alchemical region, 0 otherwise
+        )
+        energy_commands.append("do_softcore = {}".format(switch_string))
+    modified_energy_string = ";".join(energy_commands) + ";"
+    return modified_energy_string
+
+
+def force_has_softcore(energy_function):
+    """Check the energy string for indicators of softcore potentials."""
+    # Skip if the force looks like it is already alchemically modified
+    return (
+            is_variable_in_energy_string(energy_function, "lambda_electrostatics")
+            or is_variable_in_energy_string(energy_function, "lambda_sterics")
+            or is_variable_in_energy_string(energy_function, "lambda_bonds")
+            or "softcore" in energy_function
+    )
+
+
+def modify_custom_force(force, alchemical_region, do_annihilate,
+                        scale_string="lambda_custom", effective_radius_string="r", full_energy_string=None, **kwargs):
+    """
+    Alchemically modify a custom force using either a user-defined energy string or a softcore potential.
+    Alchemical parameters and other additional parameters that the modified force may depend on
+    can be specified as keyword arguments together with their default values.
+    Each parameter that begins with 'lambda_' is regarded an alchemical parameter and will be added to the alchemical
+    state when calling AlchemicalState.from_system().
+
+    Parameters
+    ----------
+    force : openmm.Force
+        The custom force (so far, custom nonbonded forces are supported,
+        as well as custom bond forces that define 1-4 interactions).
+        The force is directly modified and the original instance will be lost after calling this function.
+    alchemical_region: AlchemicalRegion
+        The alchemical region.
+    do_annihilate : bool
+        If True, annihilate the alchemical atoms, if False, decouple the alchemical region from the rest of the system.
+    scale_string : str
+        The scale factor that will be multiplied with the original energy function. This scale will usually include
+        lambda parameters.
+    effective_radius_string : str
+        The energy expression that is used for the effective radius in the softcore potential.
+    full_energy_string : str
+        When redefining the complete energy string, pass full_energy_string. This overwrites any energy expressions
+        passed as scale_string or effective_radius_string.
+    kwargs:
+        Any additional global parameters that the force depends on. Parameters that begin with "lambda_" will
+        be added to the alchemical state upon creation.
```

Comment:
I don't think this happens automatically currently. Maybe add that "will have to be added"?